services/zenoss/service.go

```diff
@@ -199,20 +199,12 @@ type Event struct {
 	TID    int64                    `json:"tid"`
 }
 
-func (s *Service) preparePost(url string, state *alert.EventState, data *alert.EventData, hc *HandlerConfig) (string, io.Reader, error) {
+func (s *Service) preparePost(state *alert.EventState, data *alert.EventData, hc *HandlerConfig) (string, io.Reader, error) {
 	c := s.config()
 	if !c.Enabled {
 		return "", nil, errors.New("service is not enabled")
 	}
 
```

Comment:
I think that it was correct to validate the URL return value in order to avoid a possible future panic, wasn't it? It would be also better to use `url.ParseRequestURI` so that relative URLs and even empty strings are not allowed. 

```suggestion

	if _, err := neturl.ParseRequestURI(c.URL); err != nil {
		return "", nil, err
	}
```

Reply:
Config is validated elsewhere, so I think it is not necessary to validate the URL here. I've replaced `Parse` with `ParseRequestURI`.
[services/zenoss/config.go#L101](https://github.com/bonitoo-io/kapacitor/blob/d7a0c037c82650fa6639ae0bcd2debc9a1198176/services/zenoss/config.go#L101)